envisage/tests/test_extension_point.py

```diff
@@ -111,6 +111,31 @@ def _x_changed(self):
         # Make sure the trait change handler was *not* called.
         self.assertEqual(False, f.x_changed_called)
 
+    def test_mutate_extension_point_no_effect(self):
+        """ Extension point is recomputed so mutation has no effect. """
+
+        registry = self.registry
+
+        # Add an extension point.
+        registry.add_extension_point(self._create_extension_point("my.ep"))
+
+        # Set the extensions.
+        registry.set_extensions("my.ep", [1, 2, 3])
+
+        # Declare a class that consumes the extension.
+        class Foo(TestBase):
+            x = ExtensionPoint(List(Int), id="my.ep")
+
+        # when
+        f = Foo()
+        f.x.append(42)
```

Comment:
Test LGTM, behavior is still a litte confusing to me.  So _effectively_ this line does nothing?

Reply:
Yep. Technically, this line does append to a list, but that list is not kept anywhere and is thrown away.